rust_src/src/crypto/mod.rs

```diff
@@ -122,20 +123,21 @@ fn get_coding_system_for_buffer(
             end.to_raw(),
             buffer_file_name(object).to_raw(),
         ];
-        let val = LispObject::from(unsafe {
+        let val = LispObject::from_raw(unsafe {
             Ffind_operation_coding_system(4, args.as_mut_ptr())
         });
         if val.is_cons() && val.as_cons_or_error().cdr().is_not_nil() {
             return val.as_cons_or_error().cdr();
         }
     }
-    if LispObject::from(buffer.buffer_file_coding_system).is_not_nil() {
+    if LispObject::from_raw(buffer.buffer_file_coding_system).is_not_nil() {
         /* If we still have not decided a coding system, use the
            default value of buffer-file-coding-system. */
-        return LispObject::from(buffer.buffer_file_coding_system);
+        return LispObject::from_raw(buffer.buffer_file_coding_system);
     }
-    let sscsf = LispObject::from(unsafe { globals.f_Vselect_safe_coding_system_function });
-    if fboundp(sscsf).is_not_nil() {
+    let sscsf =
+        LispObject::from_raw(unsafe { globals.f_Vselect_safe_coding_system_function });
+    if fboundp(LispSymbolRef::from(sscsf)).is_not_nil() {
```

Comment:
Again, `as_symbol_or_error`? Or maybe the right thing is to call `Ffboundp`. The leading `F` should take the LispObject and perform the transforms on it as if called from Lisp.

Reply:
No, `fboundp` checks if a symbol has a function definition. `boundp` checks to see if it has a value, but here we care that the value is also not nil.